tests/fpga/map_unroll_processing_elements_test.py

```diff
@@ -18,28 +18,36 @@ def test_map_unroll_processing_elements():
     gemm = importlib.util.module_from_spec(spec)
     spec.loader.exec_module(gemm)
 
+    N = 128
+    K = 256
+    M = 512
+    P = 8
+    W = 4
+    TN = 32
+    TM = 128
+
     # Create an SDFG with multiple processing elements
-    sdfg = gemm.make_sdfg("map_unroll_processing_elements", 4)
-    sdfg.specialize({"P": 4, "M": 32})
+    sdfg = gemm.make_sdfg("map_unroll_processing_elements",
+                          dace.vector(dace.float32, W))
+    sdfg.specialize({"P": P, "W": W, "TN": TN, "TM": TM})
     for state in sdfg.states():
         for node in state.nodes():
             if isinstance(node, nodes.MapEntry) and node.params == ["p"]:
                 node.unroll = False
                 node.schedule = dace.ScheduleType.Unrolled
 
     # Initialize arrays: Randomize A and B, zero C
-    A = np.ndarray([32, 32], dtype=dace.float32.type)
-    B = np.ndarray([32, 32], dtype=dace.float32.type)
-    C = np.ndarray([32, 32], dtype=dace.float32.type)
-    A[:] = np.random.rand(32, 32).astype(dace.float32.type)
-    B[:] = np.random.rand(32, 32).astype(dace.float32.type)
-    C[:] = np.random.rand(32, 32).astype(dace.float32.type)
-
-    C_regression = np.ndarray([32, 32], dtype=np.float32)
+    A = np.ndarray([N, K], dtype=dace.float32.type)
+    B = np.ndarray([K, M], dtype=dace.float32.type)
+    C = np.ndarray([N, M], dtype=dace.float32.type)
+    A[:] = np.random.rand(N, K).astype(dace.float32.type)
+    B[:] = np.random.rand(K, M).astype(dace.float32.type)
+    C[:] = np.random.rand(N, M).astype(dace.float32.type)
+
     C_regression = A @ B + C
 
-    sdfg(A=A, B=B, C=C, N=32, K=32)
-    diff = np.linalg.norm(C_regression - C) / float(32 * 32)
+    sdfg(A=A, B=B, C=C, N=N, M=M, K=K)
+    diff = np.linalg.norm(C_regression - C) / float(N * M)
```

Comment:
I think this is not robust. Better to use allclose or divide by the norm of the reference value (https://www.netlib.org/lapack/lug/node75.html)

Reply:
Fixed in 32efb3b6 to use same verification as other matmuls